src/Compilers/CSharp/Portable/Binder/Binder_Query.cs

```diff
@@ -953,25 +955,6 @@ private static UnboundLambda MakeQueryUnboundLambda(CSharpSyntaxNode node, Query
                     // Could not find an implementation of the query pattern for source type '{0}'.  '{1}' not found.
                     diagnostics.Add(ErrorCode.ERR_QueryNoProvider, node.Location, MessageID.IDS_AnonMethod.Localize(), methodName);
                 }
-                else if (ultimateReceiver.Kind == BoundKind.MethodGroup)
```

Comment:
>else if (ultimateReceiver.Kind == BoundKind.MethodGroup) [](http://example.com/codeflow?start=16&length=56)

I am not sure why this code is removed. The only difference from 'main' is the PROTOTYPE comment. Which means this code wasn't touched for extension types. I think we should keep it.

Reply:
This code is unreachable. The case above for `ultimateReceiver.HasAnyErrors` intercepts all the cases that might be handled here.
The reason I prefer to remove the code rather than just leave it is that it calls `ResolveMethodGroup` which begs the question for how extensions should be handled. But anything I do here cannot be tested.

Reply:
> This code is unreachable. The case above for ultimateReceiver.HasAnyErrors intercepts all the cases that might be handled here.

Thanks for the explanation. I am good with removal of the code.